common/roles/cert-perms/tasks/main.yml

```diff
@@ -1,111 +1,104 @@
 ---
 
-# Posix permissions
+- name: Get the certificates list
+  register: certs
+  find:
+    path: '/etc/letsencrypt/archive/{{ cert_dir }}'
+    recurse: yes
+    patterns: 'cert*,chain*,fullchain*'
 
+- name: Get the keys list
+  register: keys
+  find:
+    path: '/etc/letsencrypt/archive/{{ cert_dir }}'
+    recurse: yes
+    patterns: 'privkey*'
+
+# By default, consider the access to the private key is requested.
+# This might change to be explicitely requested by the caller task.
+- name: Set the default value of the "access_private_key" flag
+  when: access_private_key is not defined
+  set_fact:
+    access_private_key: true
```

Comment:
Does it works for the dovecot/openldap case? Or does this set_fact overrides the variable the role is called with?

Reply:
Yes, it works so far. Dovecot need to access the LDAP certificates, but not the private key. I also suspect a lot of software won't need access to the private key, but this need to be tested. Maybe Postfix and Dovecot are using a privileged account to start and downgrade their privileges for the main daemon. So this need to be tested.